### PR TITLE
chore: remove fallback for typescript compiler in legacy dir, adapt app

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR was a tough one! 😅

The issue mainly derived by the fact we had `src/legacy` in the typescript compilation paths. 
This made some imports pointing to legacy hooks and so on, to work even if they didn't start with `legacy/`.

Once I made TS to not load code from `legacy` unless you explicitly say so, there was thousands of errors. 

I manage, little by little to fix everything, however, at the end I saw that even though the web app was compiling, the test still didn't work because the absolute paths were not working. 

Long story short, at the end it was some freaking excludes in the `tsconfig`. After removing them, absolute imports worked.

# How to review
The main changes are explained in the code review below.

# To Test
- Web should work
- Tests should pass

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -8,11 +8,6 @@
   "exclude": [
     "node_modules",
     "cypress",
```
Comment (anxolin) [inline tsconfig.json:10]: Any clue why this was excluded? this was causing the absolute import issue

File: tsconfig.base.json [hunk 2]
```diff
@@ -1,8 +1,5 @@
 {
   "compilerOptions": {
-    "paths": {
-      "*": ["*", "legacy/*"]
```
Comment (anxolin) [inline tsconfig.base.json:4]: main change!

**Comment by anxolin** [comment]
Since the test are broken, I would like to merge this PR to fix it.

Let me know your comments.

File: tsconfig.json (hunk 1 above)
Comment (alfetopito) [reply tsconfig.json:10]: Because `cypress` tests are not written in TS?

Comment (anxolin) [reply tsconfig.json:10]: i left this part, i was refered to the deleted items
